nano/node/blockprocessor.cpp

```diff
@@ -376,10 +376,6 @@ nano::process_return nano::block_processor::process_one (nano::write_transaction
 				node.logger.try_log (boost::str (boost::format ("Gap previous for: %1%") % hash.to_string ()));
 			}
 			info_a.verified = result.verified;
-			if (info_a.modified == 0)
```

Comment:
I think there should be debug_asserts so `(info_a.modified != 0)`. Otherwise this may get uncovered and it is prone to insert a bug.

Reply:
This is moving the assignment of this field to right before it's inserted. Maybe "modified" should be made private and only unchecked_map can access it.

Reply:
It won't guarantee, the `modified` is already initialized as `0`. Not getting updated by a malformed construction on unchecked_map on any further change could lead to a missed case. I still think an assert wouldn't hurt in this case.

Reply:
+1 for a private `modified` member.

Can't really tell if the debug assert makes sense though, but @thsfs might have a point. Maybe we can keep the check for `release` and add the `debug` assert to catch eventual issues in debug builds?